_posts/pl/2024-05-17-czy-wiesz-ze-mozna-uniknac-nadmiernego-uzycia-warunkow-if-przez-wykorzystanie-ternary-operator-i-lookup-table.md

```diff
@@ -0,0 +1,133 @@
+---
+layout:    post
+title:     "Czy wiesz, że można uniknąć nadmiernego użycia warunków 'if' przez wykorzystanie ternary operator i lookup table?"
+date:      2024-05-17T08:00:00+01:00
+published: true
+didyouknow: true
+lang: pl
+author: wstolarski
+image: /assets/img/posts/2024-05-17-czy-wiesz-ze-mozna-uniknac-nadmiernego-uzycia-warunkow-if-przez-wykorzystanie-ternary-operator-i-lookup-table/code.webp
+tags:
+- javascript
+---
+
+Niektórzy uważają, że liczba warunków `if/else` w kodzie powinna być minimalizowana, zawsze zastępując przynajmniej pojedyncze instrukcje `if/else` operatorem ternarnym. Duża ilość warunków `if/else` często prowadzi do trudnej w zrozumieniu struktury i może przyprawić programistów o zawrót głowy.
+
+Załóżmy, że mamy kod, który na podstawie wprowadzonej wartości w polu wejściowym (`<input>`), a dokładnie marki samochodu, wyświetla jego opis.
+
+## Scenariusz z if/else
+```javascript
+const setCarDescriptionBasedOnBrand = () => {
+  const brandInput = document.getElementById('brandInput').value;
+ 
+    let carDesc = '';
+   
+    if(brandInput === 'toyota') {
+        carDesc = 'Toyota - Reliable and Fuel Efficient'
+    } else if(brandInput === 'ford') {
+        carDesc = 'Ford - American Toughness';
+    } else if(brandInput === 'honda') {
+        carDesc = 'Honda - Innovative and Efficient';
+    } else if(brandInput === 'bmw') {
+        carDesc = 'BMW - Ultimate Driving Machine';
+    } else {
+        carDesc = 'Unknown Car Brand';
+    }
+ 
+  document.getElementById('carDescription').innerText = carDesc;
+};
+```
+Choć `if/else` załatwia sprawę, jesteśmy obciążeni dużą ilością powtarzalnej logiki porównującej `brandInput` i powtarzającym się przypisywaniem `carDesc`.
```

Comment:
Choć `if/else` załatwia sprawę, to jesteśmy obciążeni dużą ilością powtarzalnej logiki porównującej `brandInput` i przypisywanym `carDesc`.